Arch: Lexical Error when trying to install xmobar from source using stack

`alsa-mixer> configure
alsa-mixer> Configuring alsa=mixer-0.3.0...
alsa-mixer> build
alsa-mixer> Preprocessing library for alsa-mixer-0.3.0..
alsa-mixer> c2hs: C header contains errors:
alsa-mixer>
alsa-mixer> .stack-work/dist/x86_64-linux-tinfo6.Cabal-3.0.1.0/build/Sound/ALSA/Mixer/Internal.i:1: (column 1) [ERROR] >>> Lexical error !
alsa-mixer> The character '#' does not fit here.
alsa-mixer>
Progress 1/2

-- While building package alsa=mixer-0.3.0 (scroll up to its section to see the error) using:
/home/cameron/.stack/setup-exe-cache/x86_64-linux-tinfo6/Cabal-simple_mPHDZzAJ_3.0.1.0_ghc-8.8.3 --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.0.1.0 build --ghc-options " -fdiagnostics-color=always"
Process exited with code: ExitFailure 1

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,5 +1,5 @@
-# ghc 8.8.3
-resolver: lts-16.0
+# ghc 8.10.7
+resolver: lts-18.20
 
 packages:
  - .

